app/js/elements/fxos-inspector.js

```diff
@@ -0,0 +1,485 @@
+      // current search text.
+      items.forEach(i => i.hidden = value && i.dataset.search.indexOf(value) === -1);
+    }, 500);
+  });
+
+  // Reset the <gaia-dialog-prompt> value when closed.
+  this.newAttributeName.addEventListener('closed', () => {
+    this.newAttributeName.els.input.value = '';
+  });
+
+  // Create a new attribute when the <gaia-dialog-prompt>
+  // is submitted.
+  this.newAttributeName.els.submit.addEventListener('click', () => {
+    try {
+      var name = this.newAttributeName.els.input.value;
+      var attribute = document.createAttribute(name);
+
+      this.target.setNamedItem(attribute);
+
+      this.dispatchEvent(new CustomEvent('createattribute', {
+        detail: JSON.stringify({
+          keyPath: this.keyPath,
+          expression: '[\'' + this.keyPath.join('\'][\'') + '\']',
+          name: name
+        })
+      }));
+    } catch (e) {
+      window.alert('Invalid attribute name');
+      return;
+    }
+
+    this.render();
+  });
+};
+
+proto.render = function() {
+  clearTimeout(this._searchTimeout);
+
+  this.target = this.rootTarget;
+  this.keyPath.forEach(key => this.target = this.target[key]);
+
+  this.container.innerHTML = renderTargetPage(this.target, this.keyPath.length === 0).innerHTML;
+  this.container.scrollTo(0, 0);
+};
+
+proto.setRootTarget = function(rootTarget) {
+  this.rootTarget = rootTarget;
+  this.target = this.rootTarget;
+
+  this.keyPath = [];
+  this.render();
+};
+
+proto._handleButtonClick = function(target) {
+  switch (target.dataset.action) {
+    case 'cancel':
+      this.keyPath.pop();
+      break;
+    case 'save':
+      (() => {
+        var value = this.container.querySelector('fxos-code-editor').value;
+        var target = this.rootTarget;
+
+        this.keyPath.forEach((key, index) => {
+          if (index < this.keyPath.length - 1) {
+            target = target[key];
+            return;
+          }
+
+          target[key] = value;
+        });
+
+        this.dispatchEvent(new CustomEvent('save', {
+          detail: JSON.stringify({
+            keyPath: this.keyPath,
+            expression: '[\'' + this.keyPath.join('\'][\'') + '\']',
+            value: value
+          })
+        }));
+
+        this.keyPath.pop();
+      })();
+      break;
+    default:
+      break;
+  }
+
+  this.render();
+};
+
+proto._handleRemoveAttributeClick = function(target) {
+  var name = target.getAttribute('href');
+  this.target.removeNamedItem(name);
+
+  this.dispatchEvent(new CustomEvent('removeattribute', {
+    detail: JSON.stringify({
+      keyPath: this.keyPath,
+      expression: '[\'' + this.keyPath.join('\'][\'') + '\']',
+      name: name
+    })
+  }));
+
+  this.render();
+};
+
+proto._handleListItemClick = function(target) {
+  // If the link clicked has an `href` pointing
+  // to "../", go back to the previous level in
+  // the key path.
+  var href = target.getAttribute('href');
+  if (href === '../') {
```

Comment:
For better user experience, I think we should write "Up one level" instead of "..".